x/liquidstakeibc/keeper/delegation_strategy.go

```diff
@@ -46,28 +63,53 @@ func (k *Keeper) GenerateDelegateMessages(hc *types.HostChain, depositAmount sdk
 
 	messages := make([]proto.Message, 0)
 	for _, delegationAmount := range delegateAmounts {
-		message := &stakingtypes.MsgDelegate{
-			DelegatorAddress: hc.DelegationAccount.Address,
-			ValidatorAddress: delegationAmount.ValAddress,
+		var message proto.Message
+		if !undelegating {
```

Comment:
This creates a mess to debug too, can you just entirely separate these functions? delegate different, undelegation different?

Reply:
There are only 2 or 3 different lines in he whole function code. It is very similar to what we have in `lscomsos`

Reply:
okay!